### PR TITLE
Introduce new property 'subzero.referenceresolver.class' where we could specify Kryo's ReferenceResolver implementation

### PR DESCRIPTION
**Pull request by acieplak**

### CODE REVIEW

**Comment by acieplak** [comment]
Hi,

It's rather an enhancement request, but in KryoStrategy#newKryoInstance() we create a Kryo with default MapReferenceResolver. I have found out that in some cases this implementation is a lot more slowly than others, e.g. ListReferenceResolver. The difference sometimes is about 10 times, which is a lot.
It would be nice to have some kind of system property, e.g. a subzero.referenceresolver.class where we could specify which ReferenceResolver implementation should be used.

**Comment by acieplak** [comment]
Hi, this PR is a followup to the issue [#24](https://github.com/jerrinot/subzero/issues/24)

File: subzero-core/src/main/java/info/jerrinot/subzero/internal/strategy/KryoStrategy.java [hunk 1]
```diff
@@ -41,10 +47,10 @@ private Kryo newKryoInstance() {
             kryo = new Kryo();
         } else {
             ClassLoader classLoader = ClassLoaderUtils.getConfiguredClassLoader(hazelcastInstance);
```
Comment (jerrinot) [inline subzero-core/src/main/java/info/jerrinot/subzero/internal/strategy/KryoStrategy.java]: shouldnt you respect the properties even when the `IGNORE_HAZELCAST_CLASSLOADER` is set to `true` ?

Comment (acieplak) [reply subzero-core/src/main/java/info/jerrinot/subzero/internal/strategy/KryoStrategy.java]: That's a good question. We could assume that the property `com.esotericsoftware.kryo.util.MapReferenceResolver` should ignore whether the Hazelcast class loader is ignored or not as we care only about the `ReferenceResolver`.
So in line 42 we should have sth like this:
```
kryo = new Kryo(createReferenceResolver());
```
Do you agree?

Comment (jerrinot) [reply subzero-core/src/main/java/info/jerrinot/subzero/internal/strategy/KryoStrategy.java]: yes. :point_up: looks good.
thanks for the PR!

**Comment by acieplak** [comment]
Hi,
Could you let me know if you would like to merge this pr and accordingly release a new version? And if so when could I expect this to happen?
Best regards,
Adrian

**Comment by jerrinot** [comment]
thanks for the PR, I will push a new release out today!

**Comment by jerrinot** [comment]
@acieplak: I just released 0.8 version with your fix. it should appear in Maven Central in a short while.

**Pull request merged**